### PR TITLE
[Ot171-39] Slider Testimonios

### PR DESCRIPTION
**Pull request by frannasich**
### Slider Testimonios changes description:
-  ScrollView added and slider Testimonios added
-  Added cases of slides
-  Added TestimoniosCollectionViewCell
-  Some images added
-  Testimonios max 4 pages added
-  Added SeeMoreCollectionViewCell

https://user-images.githubusercontent.com/80861248/161852712-b4bd7fbc-a22c-452b-9d4b-0a5a99b585c9.mov

**Jira Ticket**
[OT171-39](https://alkemy-labs.atlassian.net/jira/software/c/projects/OT171/boards/247?modal=detail&selectedIssue=OT171-39)

### CODE REVIEW

File: SomosMasApp/SomosMasApp/App/Shared/Modules/Home/Controller/TestimoniosCollectionViewCell.swift [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+//  SomosMasApp
+//
+//  Created by Fran Nasich on 01/04/2022.
+//
+
+import UIKit
+
+class TestimoniosCollectionViewCell: UICollectionViewCell {
+
+    @IBOutlet weak var testimonioImage: UIImageView!
```
Comment (gapoj) [inline SomosMasApp/SomosMasApp/App/Shared/Modules/Home/Controller/TestimoniosCollectionViewCell.swift:12]: naming is an important thing, all the app is in English why you name your clases in Spanish???? remember that this is a public repository any one can see it

File: SomosMasApp/SomosMasApp/App/Shared/Modules/Home/View/TestimoniosCollectionViewCell.xib [hunk 2]
```diff
@@ -0,0 +1,90 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<document type="com.apple.InterfaceBuilder3.CocoaTouch.XIB" version="3.0" toolsVersion="20037" targetRuntime="iOS.CocoaTouch" propertyAccessControl="none" useAutolayout="YES" useTraitCollections="YES" useSafeAreas="YES" colorMatched="YES">
```
Comment (gapoj) [inline SomosMasApp/SomosMasApp/App/Shared/Modules/Home/View/TestimoniosCollectionViewCell.xib:2]: Again naming in Spanish

File: SomosMasApp/SomosMasApp/App/Shared/Modules/Home/Controller/HomeViewController.swift [hunk 3]
```diff
@@ -33,30 +48,54 @@ class HomeViewController: UIViewController {
         collectionView.contentInset = UIEdgeInsets(top: 0, left: 0, bottom: 0, right: 0)
         collectionView.register(UINib(nibName: "HomeCollectionViewCell", bundle: nil), forCellWithReuseIdentifier: "mycell")
         
+        testimoniosCollectionView.isPagingEnabled = true
+        testimoniosCollectionView.dataSource = self
+        testimoniosCollectionView.delegate = self
+        testimoniosCollectionView.contentInset = UIEdgeInsets(top: 0, left: 0, bottom: 0, right: 0)
```
Comment (gapoj) [inline SomosMasApp/SomosMasApp/App/Shared/Modules/Home/Controller/HomeViewController.swift:54]: why do you add an inset if you didn't modify anything

File: SomosMasApp/SomosMasApp/App/Shared/Modules/Home/Controller/HomeViewController.swift [hunk 4]
```diff
@@ -33,30 +48,54 @@ class HomeViewController: UIViewController {
...
 extension HomeViewController: UICollectionViewDataSource {
     
     func numberOfSections(in collectionView: UICollectionView) -> Int {
         return 1
     }
-    
+
     func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
-        return sliderData.count
+        switch collectionView {
```
Comment (gapoj) [inline SomosMasApp/SomosMasApp/App/Shared/Modules/Home/Controller/HomeViewController.swift:68]: Good, I like this

File: SomosMasApp/SomosMasApp/App/Shared/Modules/Home/Controller/TestimoniosCollectionViewCell.swift [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+    @IBOutlet weak var testimonioEpigrafe: UILabel!
+    @IBOutlet weak var verMasButton: UIButton!
+
```
Comment (gapoj) [inline SomosMasApp/SomosMasApp/App/Shared/Modules/Home/Controller/TestimoniosCollectionViewCell.swift:15]: Why are you using a button???? a button is never the solution on a cell, you should use the delegates like in tableviews

File: SomosMasApp/SomosMasApp/App/Shared/Modules/Home/Controller/HomeViewController.swift [hunk 6]
```diff
@@ -33,30 +48,54 @@ class HomeViewController: UIViewController {
...
+        case self.testimoniosCollectionView:
+            // Return Max pages = 4
+            return testimoniosData.count > 4 ? 4 : testimoniosData.count
```
Comment (gapoj) [inline SomosMasApp/SomosMasApp/App/Shared/Modules/Home/Controller/HomeViewController.swift:71]: read the history in jira " Este slider deberá tener una capacidad máxima de 4 y como último elemento una flecha que dará acción a abrir una nueva pantalla (no es objetivo de esta historia)."  thats 4 elements plus the arrow.  You ask and I literally said that is anotar item,  and you should not ever use a button in a cell when you can react to the selection

**Review by gapoj — CHANGES_REQUESTED** [review]
the naming is important but not the main reason because I ask for changes(change it anyway), the button is a conceptual error and you miss the acceptance criteria

File: SomosMasApp/SomosMasApp/App/Shared/Modules/Home/Controller/HomeViewController.swift (hunk 6 above)
Comment (frannasich) [reply SomosMasApp/SomosMasApp/App/Shared/Modules/Home/Controller/HomeViewController.swift:71]: But that i did with the four pages is okey right?

File: SomosMasApp/SomosMasApp/App/Shared/Modules/Home/Controller/TestimoniosCollectionViewCell.swift (hunk 1 above)
Comment (frannasich) [reply SomosMasApp/SomosMasApp/App/Shared/Modules/Home/Controller/TestimoniosCollectionViewCell.swift:12]: Ok

**Review by gapoj — CHANGES_REQUESTED** [review]
PR WITH CONFLICTS

**Comment by frannasich** [comment]
> PR WITH CONFLICTS

What conflict?

**Comment by gapoj** [comment]
> > PR WITH CONFLICTS
> 
> What conflict?

You should be able to identify that GitHub literally saids: "This branch has conflicts that must be resolved"

**Comment by gapoj** [comment]
the author made a different PR for this

**Pull request closed without merging**